### PR TITLE
Implement Updater

### PR DESCRIPTION
**Pull request by SparklingComet**
Checks for updates upon enabling the plugin.

Todo:
- [x] Config setting to disable the updater.
- [x] Testing

### CODE REVIEW

**Comment by KillerOfPie** [comment]
Looks good to me, I'll probably have time to test it on Thursday if you don't by then.

**Review by KillerOfPie — APPROVED** [review]

**Comment by SparklingComet** [comment]
I won't be able to test it as I'll still be out not town.
I have typed all of this on my tablet so there could be missing imports etc.

**Comment by KillerOfPie** [comment]
@SparklingComet should be good now only had 2 issues

File: pom.xml [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
     <groupId>org.shanerx</groupId>
     <artifactId>tradeshop</artifactId>
-    <version>1.6.0-BETA</version>
+    <version>1.6.0-Updater</version>
```
Comment (SparklingComet) [inline pom.xml]: What's this? This isn't a valid release tag. I assume you wanted to check wether the latest build had been compiled...?
If yes, this should be changed back to BETA.

File: src/main/java/org/shanerx/tradeshop/Updater.java [hunk 2]
```diff
@@ -62,11 +64,19 @@
...
 
 	public Updater(PluginDescriptionFile pdf) {
 		this.pdf = pdf;
+
+		log = Bukkit.getLogger();
+
+		try {
+			url = new URL("https://api.spigotmc.org/legacy/update.php?resource=32762"); // Edit API URL.
+		} catch (MalformedURLException ex) {
+			log.log(Level.WARNING, "Error: Bad URL while checking {0} !", pdf.getName());
```
Comment (SparklingComet) [inline src/main/java/org/shanerx/tradeshop/Updater.java:78]: Maybe use String.format(String, String)

Btw, I completely forgot about that try..catch block.

File: src/main/java/org/shanerx/tradeshop/Utils.java [hunk 3]
```diff
@@ -21,7 +21,11 @@
 
 package org.shanerx.tradeshop;
 
-import org.bukkit.*;
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.Material;
+import org.bukkit.Nameable;
+import org.bukkit.OfflinePlayer;
```
Comment (SparklingComet) [inline src/main/java/org/shanerx/tradeshop/Utils.java:28]: Good one, thought we had ironed out all wildcard imports.

File: src/main/resources/plugin.yml [hunk 4]
```diff
@@ -1,5 +1,5 @@
 name: TradeShop
-version: 1.6.0-BETA
+version: 1.6.0-Updater
```
Comment (SparklingComet) [inline src/main/resources/plugin.yml]: Copying from above:

> What's this? This isn't a valid release tag. I assume you wanted to check wether the latest build had been compiled...?
If yes, this should be changed back to BETA.

File: src/main/java/org/shanerx/tradeshop/Updater.java (hunk 2 above)
Comment (SparklingComet) [reply src/main/java/org/shanerx/tradeshop/Updater.java:78]: Also, I think we could also use an empty catch block as that URL is a constant and is definitely a valid URL.

File: pom.xml (hunk 1 above)
Comment (KillerOfPie) [reply pom.xml]: Correct, and I edited and pushed right before I left my house should already be fixed

File: src/main/java/org/shanerx/tradeshop/Updater.java (hunk 2 above)
Comment (KillerOfPie) [reply src/main/java/org/shanerx/tradeshop/Updater.java:78]: Should always have an error output just incase, it should never but that doesn't mean it won't and that is the proper way for the url to be done.

File: src/main/java/org/shanerx/tradeshop/Utils.java (hunk 3 above)
Comment (KillerOfPie) [reply src/main/java/org/shanerx/tradeshop/Utils.java:28]: I didn't even touch that, so my ide auto corrected that

File: src/main/resources/plugin.yml (hunk 4 above)
Comment (KillerOfPie) [reply src/main/resources/plugin.yml]: Same as above

**Comment by SparklingComet** [comment]
If you have tested it, it should be ready for merge.

**Comment by KillerOfPie** [comment]
I've tested it to the point that it tells me im ahead when im working on things

**Comment by SparklingComet** [comment]
I assume I will merge it with ``1.6`` then.
I will not delete the ``Updater`` branch yet, though.

**Pull request merged**